First draft add ONT to fc trends plot

sofiahag opened the pull request:
We will need to discuss which values we want displayed for ONT. I found something called "Yield passed" and "Read count passed". I don't know enough about ONT to know what these represent 🤷🏻‍♀️ 

Currently we have total clusters and total yield per flowcell or lane. I added the read count passed as the default value, and read count passed and yield passed to be displayed when only either PromethION 23 or MN19414 are being selected (on tooltip hover).

I'm not sure how these instruments differ either or how they should be displayed, or even if they should be displayed. But I made a draft version so we have something to look at for the meeting. I'll make this a WIP.

Also, the chemistry colors are, as usual, acting up, so I had it, and hard coded the colors. This should be cleaned up a bit.

sofiahag commented:
Up on stage!

kedhammar commented:
Right it looks as though the date filtering does not work, and that the Illumina vs ONT flow cells are segregated on the X-axis.

It could be nice to visualize the ONT flow cell trends, but I think there is very little overlap between which metadata metrics are relevant for Illumina and ONT? Is there a point to showing them in the same plot with the same filtering parameters?

sofiahag commented:
> Right it looks as though the date filtering does not work, and that the Illumina vs ONT flow cells are segregated on the X-axis.
> 
> It could be nice to visualize the ONT flow cell trends, but I think there is very little overlap between which metadata metrics are relevant for Illumina and ONT? Is there a point to showing them in the same plot with the same filtering parameters?

Yes, I didn't check all the functionalities, I thought we need to decide how we want to display the data first. 

Ok, I see! Yes, I haven't been involved in ONT at all, but then I know they are too different to even include in the same plot. It will be easier to implement a new plot also, great, then I know, thank you for your input! :-)

sofiahag commented:
Ok, I created a new plot for the ONT FC's instead, I don't know if you guys think it should be featured on the same page (FC trends), further down instead? I added some data, not sure what we want to display, please have a look here: 
https://genomics-status-stage.scilifelab.se/ont_flowcells_plot

kedhammar commented:
Cool!

I think most of the quantitative metrics shown on https://genomics-status.scilifelab.se/flowcells would be nice to be able to visualize, but I'm sure the ONT lab peeps will have ideas as well.

sofiahag commented:
Right! I think project name would be good to have, is there only one sample per flowcell? In that case the sample name could be included as well.

sofiahag commented:
Ah, ok, so maybe not that information then. Ok, I can ask the lab/Ellen and see what they would like to be included in the plot 👍🏻

sofiahag commented:
I don't know if we should interpret no feedback as approval in this case, and just go with this, and wait for people to come back with feedback once it's been deployed? 😛 That's when you usually get feedback 😆

kedhammar commented:
> I don't know if we should interpret no feedback as approval in this case, and just go with this, and wait for people to come back with feedback once it's been deployed? 😛 That's when you usually get feedback 😆

Personally I think it would be good if the X-axis (time) was properly scaled, instead of having equidistant datapoints, so that we can follow the ONT throughput over time.

kedhammar commented:
Maybe also having prefixed units (Kb, Mb, Gb, etc) for the quantitative data that is shown, because looking at 9-figure integers is a bit tricky for my brain at least 😆

sofiahag commented:
> > I don't know if we should interpret no feedback as approval in this case, and just go with this, and wait for people to come back with feedback once it's been deployed? 😛 That's when you usually get feedback 😆
> 
> Personally I think it would be good if the X-axis (time) was properly scaled, instead of having equidistant datapoints, so that we can follow the ONT throughput over time.

We're using this time scale in the original flowcells trend plot, so I guessed it's something we're comfortable with, but maybe the ONT would benefit from another scale, I don't know, what do you think @aanil @alneberg ?

sofiahag commented:
> Maybe also having prefixed units (Kb, Mb, Gb, etc) for the quantitative data that is shown, because looking at 9-figure integers is a bit tricky for my brain at least 😆

Once again, this is the formatting we're currently using for the default FC plot. In that case maybe we should re-format it for both plots? I haven't heard any complaints from the people looking at the original FC plot though

alneberg commented:
If it's possible to have a switch between the two different types of x-axes then that would be nice?

alneberg commented:
They serve different purposes since if you have a lot of lanes at the same day (like with 10B illumina flowcells) you would like to be able to tell them apart.

sofiahag commented:
Ok, so there could be a lot of runs on the same day? Right, then I understand that we want to use a different scale, I didn't know that. Ok, I'll look in to it! 👍🏻

sofiahag commented:
Something like this? https://genomics-status-stage.scilifelab.se/ont_flowcells_plot

sofiahag commented:
Or did you mean grouping data points together for the same date, maybe?

sofiahag commented:
Oh wait, I see the data points are not aligning with the x-axis properly. I remember us having this issue with the default FC trends plot also 🤔

sofiahag commented:
I think all the data should now be displayed chronologically. However Highcharts doesn't like the time format so it reverts to using raw timestamp values in milliseconds. I removed the labels now. I guess I could continue working on it...

sofiahag commented:
Ok, I got the dates to be formatted properly now!

sofiahag commented:
I added formatting of the values now, please see if this is what you were looking for, @kedhammar 😄

alneberg commented:
The x-axis look good now! Do you think it would be difficult to add another switch for the Y-axis so one could have either number of reads (as it is now) and the amount of bases sequenced? Also, maybe @kedhammar could say whether we should have the total number of reads (as it is now) or if we should use the ones that pass filter only instead?

Also, I don't think it should say "MB" after the number of reads, but only "M": 
<img width="286" alt="Screenshot 2024-01-22 at 07 49 55" src="https://github.com/SciLifeLab/genomics-status/assets/1250075/91182578-5e5c-4608-8be7-3ddf78156a36">

sofiahag commented:
> The x-axis look good now! Do you think it would be difficult to add another switch for the Y-axis so one could have either number of reads (as it is now) and the amount of bases sequenced? Also, maybe @kedhammar could say whether we should have the total number of reads (as it is now) or if we should use the ones that pass filter only instead?
> 
> Also, I don't think it should say "MB" after the number of reads, but only "M": <img alt="Screenshot 2024-01-22 at 07 49 55" width="286" src="https://private-user-images.githubusercontent.com/1250075/298487438-91182578-5e5c-4608-8be7-3ddf78156a36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MTMzMTYsIm5iZiI6MTcwNTkxMzAxNiwicGF0aCI6Ii8xMjUwMDc1LzI5ODQ4NzQzOC05MTE4MjU3OC01ZTVjLTQ2MDgtOGJlNy0zZGRmNzgxNTZhMzYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjJUMDg0MzM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzgzMmZiNmY0NDkzZGU2MTkxN2MxNzIyN2MwMzMzN2EyOTE0MGIyNDgxNDdiZWUwYTk4NjY4YzUwZTVkN2JjMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ZlJ_FOvEBmwMlnaArRjNg864MVh1Lc0wBd6OVQKA3bE">

Sure! That would make sense. Would you say that "read count passed" and "basecalled pass bases" would be the values to use then, for the y-axis?

sofiahag commented:
Or maybe "basecalled pass read count"? I'm currently using "read count passed" for the y-axis...

sofiahag commented:
Ok, I implemented your suggested changes, @alneberg!

I added the "read count passed" and "base count passed" now, not sure if these are the values we want?

alneberg commented on run_dir/static/js/ont_trend_plot.js:
```diff
@@ -133,6 +106,38 @@ function make_plot(key, name, filter_inst_type, color_type, plot_type){
+            const color = this.point.color;
+            const series_name = this.series.name || '';
+            const y_value = typeof this.y !== 'undefined' ? this.y : '';
+            const passed_bases = typeof this.point.passed_bases !== 'undefined' ? format_value(this.point.passed_bases) : '';
```
Could we have passed_reads here as well, and passed bases correspondingly within the basecalled_pass_bases above? I think it would be good.

alneberg reviewed (approved):
I like it! Thank you!
